StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1207CodeFixProvider.cs

```diff
@@ -0,0 +1,83 @@
+﻿namespace StyleCop.Analyzers.OrderingRules
+{
+    using System.Collections.Immutable;
+    using System.Composition;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CodeActions;
+    using Microsoft.CodeAnalysis.CodeFixes;
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+    using StyleCop.Analyzers.Helpers;
+
+    /// <summary>
+    /// Implements code fixes for <see cref="SA1207ProtectedMustComeBeforeInternal"/>.
+    /// </summary>
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(SA1207CodeFixProvider))]
+    [Shared]
+    public class SA1207CodeFixProvider : CodeFixProvider
+    {
+        private static readonly ImmutableArray<string> FixableDiagnostics =
+            ImmutableArray.Create(SA1207ProtectedMustComeBeforeInternal.DiagnosticId);
+
+        /// <inheritdoc/>
+        public override ImmutableArray<string> FixableDiagnosticIds => FixableDiagnostics;
+
+        /// <inheritdoc/>
+        public override FixAllProvider GetFixAllProvider()
+        {
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        /// <inheritdoc/>
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            foreach (Diagnostic diagnostic in context.Diagnostics.Where(d => FixableDiagnostics.Contains(d.Id)))
+            {
+                context.RegisterCodeFix(CodeAction.Create(OrderingResources.SA1207CodeFix, token => GetTransformedDocumentAsync(context.Document, diagnostic, token)), diagnostic);
+            }
+
+            return SpecializedTasks.CompletedTask;
+        }
+
+        private static async Task<Document> GetTransformedDocumentAsync(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var originalDeclarationNode = syntaxRoot.FindNode(diagnostic.Location.SourceSpan) as MemberDeclarationSyntax;
+
+            var childTokens = originalDeclarationNode?.ChildTokens();
+            if (childTokens == null)
+            {
+                return document;
+            }
+
+            var newDeclarationNode = originalDeclarationNode.ReplaceTokens(childTokens, ComputeReplacementToken);
+
+            var newSyntaxRoot = syntaxRoot.ReplaceNode(originalDeclarationNode, newDeclarationNode);
+            return document.WithSyntaxRoot(newSyntaxRoot);
+        }
+
+        // QUESTION: Why two tokens as argument?
+        // QUESTION: Which token should we use?
```

Comment:
* Sometimes we use a `Dictionary<SyntaxToken, SyntaxToken>` to map nodes we analyze to their replacements. In this case, we want to use `originalToken` because that's the key we added to the dictionary.
* Sometimes we want to create a new token right within this method, as a function of the current input to the method. In this case, we want to use `rewrittenToken` because that's the token in the most recent state of the tree.